### ISSUE TITLE
ClassifierOutputReST couples gradients across the batch via default cross_entropy reduction

### ISSUE THREAD

**Issue by shaun0927** [jacobgil/pytorch-grad-cam#584]
While using `ClassifierOutputReST` (introduced alongside ShapleyCAM in #550) on a batched input, I noticed the saliency map for one sample depended on the other samples in the same batch. Tracked it down to the default reduction of `F.cross_entropy`.

### Code (`pytorch_grad_cam/utils/model_targets.py:37-39`)

```python
target = torch.tensor([self.category] * model_output.shape[0], device=model_output.device)
return model_output[:, self.category] - torch.nn.functional.cross_entropy(model_output, target)
```

`F.cross_entropy` defaults to `reduction='mean'`, so the right-hand side is a *scalar averaged across the batch*. Subtracting it from the per-sample logits gives correct values but couples each sample's gradient to every other sample through that mean.

### Repro

```python
import torch
from pytorch_grad_cam.utils.model_targets import ClassifierOutputReST

batch = torch.randn(4, 10, requires_grad=True)
out1 = ClassifierOutputReST(3)(batch)

batch2 = batch.detach().clone().requires_grad_(True)
batch2.data[0, 5] += 1.0            # perturb only sample 0
out2 = ClassifierOutputReST(3)(batch2)

print((out2 - out1.detach()).abs())
# tensor([0.0158, 0.0158, 0.0158, 0.0158])  -- samples 1-3 shifted too
```

If the samples were truly independent only the first entry would move; here every sample moves by exactly the mean-CE contribution.

### Proposed fix

`reduction='none'` and index `self.category` per sample:

```python
ce = torch.nn.functional.cross_entropy(model_output, target, reduction='none')
return model_output[:, self.category] - ce
```

Matches the paper's per-sample formulation (`arxiv:2501.06261`). Happy to open a one-line PR if that's how you'd like to handle it.
